onnxruntime/core/framework/utils.cc

```diff
@@ -502,6 +530,71 @@ static common::Status ExecuteGraphImpl(const SessionState& session_state,
   return Status::OK();
 }
 
+#ifdef ENABLE_TRAINING
+static common::Status ExecuteGraphImpl(const SessionState& session_state,
+                                       const FeedsFetchesManager& feeds_fetches_manager,
+                                       const std::vector<OrtValue>& feeds, std::vector<OrtValue>& fetches,
+                                       const std::unordered_map<size_t, IExecutor::CustomAllocator>& fetch_allocators,
+                                       ExecutionMode execution_mode, const bool& terminate_flag,
+                                       const logging::Logger& logger, PartialGraphExecutionState& state, const bool only_execute_path_to_fetches = false) {
```

Comment:
Consider removing the `execution_mode` argument, since we enforce it being sequential (and aren't likely to add a parallel implementation soon).

Also consider removing the `only_execute_path_to_fetches` argument, since I'm not convinced that it's compatible with the partial run idea, and it seems tricky to test.

And `fetch_allocators`, since it's always passed an empty map.